libafl/src/stages/mod.rs

```diff
@@ -446,48 +462,51 @@ where
     }
 
     #[inline]
-    fn restart_progress_should_run(&mut self, _state: &mut Self::State) -> Result<bool, Error> {
+    fn should_run(&mut self, state: &mut Self::State) -> Result<bool, Error> {
         // TODO: Proper restart handling - call post_exec at the right time, etc...
-        Ok(true)
+        RestartHelper::no_retry(state, &self.name)
     }
 
     #[inline]
-    fn clear_restart_progress(&mut self, _state: &mut Self::State) -> Result<(), Error> {
-        Ok(())
+    fn clear_progress(&mut self, state: &mut Self::State) -> Result<(), Error> {
+        RestartHelper::clear_progress(state, &self.name)
     }
 }
 
 /// Progress which permits a fixed amount of resumes per round of fuzzing. If this amount is ever
 /// exceeded, the input will no longer be executed by this stage.
 #[derive(Clone, Deserialize, Serialize, Debug)]
-pub struct RetryRestartHelper {
+pub struct RestartHelper {
```

Comment:
We have other RestartHelpers like the ExecutionCountRestartHelper, so this name is underspecified, `RetryRestartHelper::no_retry` is fine

Reply:
Or `RetryRestartHelper::none()`

Reply:
how about StdRestartHelper?

to me, "restart for up to X times" is the most standard way to retry things

Reply:
RetryRestartHelper contains two words Restart and Retry. but these two means almost the same thing